src/VirtualClient/VirtualClient.Actions/ApacheBench/ApacheBenchExecutor.cs

```diff
@@ -0,0 +1,238 @@
+                        .ConfigureAwait(false);
+                }
+
+                this.WorkloadExecutablePath = "/usr/bin/ab";
+            }
+
+            state.ApacheBenchStateInitialized = true;
+            await this.stateManager.SaveStateAsync<ApacheBenchState>($"{nameof(ApacheBenchState)}", state, cancellationToken);
+        }
+
+        private Task<string> ExecuteWorkloadAsync(string commandArguments, EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            EventContext relatedContext = telemetryContext.Clone()
+                .AddContext("packageName", this.PackageName)
+                .AddContext("command", this.WorkloadExecutablePath)
+                .AddContext("commandArguments", commandArguments);
+
+            return this.Logger.LogMessageAsync($"{nameof(ApacheBenchExecutor)}.ExecuteWorkload", relatedContext, async () =>
+            {
+                using (BackgroundOperations profiling = BackgroundOperations.BeginProfiling(this, cancellationToken))
+                {
+                    using (IProcessProxy workloadProcess = this.processManager.CreateProcess(this.WorkloadExecutablePath, commandArguments))
+                    {
+                        this.CleanupTasks.Add(() => workloadProcess.SafeKill());
+                        await workloadProcess.StartAndWaitAsync(cancellationToken)
+                            .ConfigureAwait(false);
+
+                        if (!cancellationToken.IsCancellationRequested)
+                        {
+                            await this.LogProcessDetailsAsync(workloadProcess, telemetryContext, nameof(ApacheBenchExecutor), logToFile: true)
+                                .ConfigureAwait(false);
+
+                            workloadProcess.ThrowIfErrored<WorkloadException>(ProcessProxy.DefaultSuccessCodes, errorReason: ErrorReason.WorkloadFailed);
+                            if (workloadProcess.StandardOutput.Length == 0)
+                            {
+                                throw new WorkloadException(
+                                    $"Unexpected workload results outcome. The workload did not produce any results to standard output.",
+                                    ErrorReason.WorkloadResultsNotFound);
+                            }
+                        }
+
+                        return workloadProcess.StandardOutput.ToString();
+                    }
+                }
+            });
+        }
+
+        private Task CaptureMetricsAsync(string results, string commandArguments, DateTime startTime, DateTime endTime, EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                results.ThrowIfNullOrWhiteSpace(nameof(results));
+
+                this.Logger.LogMessage($"{nameof(ApacheBenchExecutor)}.CaptureMetrics", telemetryContext.Clone()
+                    .AddContext("results", results));
+
```

Comment:
could u plz add scenario in metadata contract - [https://github.com/microsoft/VirtualClient/blob/021b71faccbe94ee886b31e68ec1585cba707128/src/VirtualClient/VirtualClient.Actions/Gzip/GzipExecutor.cs#L183 (Opens in new window or tab)](https://github.com/microsoft/VirtualClient/blob/021b71faccbe94ee886b31e68ec1585cba707128/src/VirtualClient/VirtualClient.Actions/Gzip/GzipExecutor.cs#L183)